pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -194,16 +196,20 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 		return results.WithError(err)
 	}
 
-	if esReachable {
-		// reconcile the license
-		if err := license.Reconcile(ctx, d.Client, d.ES, esClient); err != nil {
-			d.ReconcileState.AddEvent(
-				corev1.EventTypeWarning,
-				events.EventReasonUnexpected,
-				fmt.Sprintf("Could not update cluster license: %s", err.Error()),
-			)
-			log.Info("Could not update cluster license", "err", err, "namespace", d.ES.Namespace, "es_name", d.ES.Name)
-			// don't error out the entire reconciliation, move on with next steps and retry later
+	if esReachable { //nolint:nestif
+		// reconcile the Elasticsearch license
+		supportedDistribution, err := license.Reconcile(ctx, d.Client, d.ES, esClient)
+		if err != nil {
+			if !supportedDistribution {
```

Comment:
I think we can remove one level of nesting here:
```golang
supportedDistribution, err := license.Reconcile(ctx, d.Client, d.ES, esClient)
if !supportedDistribution {} // implies err != nil but we add that to condition to make it more explicit
if err != nil {}
```

Reply:
Indeed, it's better but I'd stick to the explicit way.
```
supportedDistribution, err := license.Reconcile(ctx, d.Client, d.ES, esClient)
if err != nil && !supportedDistribution {}
if err != nil {}
```